tasks/taxon_id/freyja/task_freyja.wdl

```diff
@@ -116,16 +116,46 @@ task freyja_one_sample {
   #Output QC values to the Terra data table
   python <<CODE
   import csv
+  import pandas as pd
+  
   #Want coverage output from freyja_demixed tsv file
   with open("~{samplename}_freyja_demixed.tsv",'r') as tsv_file:
     tsv_reader = csv.reader(tsv_file, delimiter="\t")
     for line in tsv_reader:
       if "coverage" in line[0]:
         with open("COVERAGE", 'wt') as coverage:
           coverage.write(line[1])
-
+      if "lineages" in line[0]:
+              with open("LINEAGES", 'wt') as lineages:
+                lineages.write(line[1].replace(" ", ","))
+      if "abundances" in line[0]:
+        with open("ABUNDANCES", 'wt') as abundances:
+          abundances.write(line[1].replace(" ", ","))
+      if "resid" in line[0]:
+        with open("RESID", 'wt') as resid:
+          resid.write(line[1])
+      if "summarized" in line[0]:
+        with open("SUMMARIZED", 'wt') as summarized:
+          summarized.write(line[1])
+  
+  dataf = pd.read_csv("~{samplename}_freyja_demixed.tsv", sep="\t", header=None)
+  dataf.columns = ["Attribute", "~{samplename}"]
+  print(dataf)
+  # Initialize a list to store output rows
+  output_data = []
+  parsed_data = {
+    "LIMS_ID": "~{samplename}",
+    "summarized": dataf.loc[dataf['Attribute'] == "summarized", "~{samplename}"].values[0],
```

Comment:
you are checking for errors above. does this work when the anticipated column does not exist?

Reply:
I moved it up to ensure that the values exist before being populated into the dictionary. This will solve the scenario when a column does not exist (it just won't show up in the final file). The wrangled output file looks exactly the same. 
![image](https://github.com/user-attachments/assets/47e55886-b2fc-4dc7-8b61-8aa42bf1427c)

Reply:
i like this much better 👍